### PR TITLE
JS solution for dojo 20221011

### PR DESCRIPTION
**Pull request by brehberg**
# Language

What language was used for the solution? 
 Javascript

# Challenge

What is the name of the challenge being completed?
Find Three Largest Numbers

### CODE REVIEW

File: challenges/2022-10-11-blast-from-the-past/solutions/javascript/brehberg/brehberg.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// - You may not sort the input array
+// - The function should handle duplicate integers; for example `[10, 5, 9, 10, 12]`
+//   should return `[10, 10, 12]`
+// - Constant space -> you will return a new array of 3 integers, and this will be the
+//   only new data structure you create.
+// - Linear time -> you should solve this problem by only passing through the array a
+//   single time.
+
+function findThreeLargestNumbers(arr) {
+  if (arr.length < 2) return null;
```
Comment (drkennetz) [inline challenges/2022-10-11-blast-from-the-past/solutions/javascript/brehberg/brehberg.js:20]: I think this should be arr.length < 3 right? length are usually 1 based

**Review by drkennetz — CHANGES_REQUESTED** [review]
Hadn't seen this implementation before, nice! 

the early exit condition I think should be arr.length < 3.

File: challenges/2022-10-11-blast-from-the-past/solutions/javascript/brehberg/brehberg.js (hunk 1 above)
Comment (drkennetz) [reply challenges/2022-10-11-blast-from-the-past/solutions/javascript/brehberg/brehberg.js:20]: Means we have poor test cases :D

Comment (brehberg) [reply challenges/2022-10-11-blast-from-the-past/solutions/javascript/brehberg/brehberg.js]: Correct on both counts :) 
My solution has been updated and a few failing JS tests were added

File: challenges/2022-10-11-blast-from-the-past/setups/javascript/solution.test.js [hunk 2]
```diff
@@ -21,3 +21,15 @@ test("Test Case Three", () => {
   const actual = findThreeLargestNumbers(input);
   expect(actual).toBeNull();
 });
+
+test("Boundary value test for input length", () => {
```
Comment (drkennetz) [inline challenges/2022-10-11-blast-from-the-past/setups/javascript/solution.test.js:25]: Thanks for these 👍

**Review by drkennetz — DISMISSED** [review]

**Comment by brehberg** [comment]
It seems that this solution doesn't actually work.  The existing tests pass, but the returned arrays were not correct.  
In the assert part of the unit tests, we need to use` .toBe(true);` instead of `.toBeTruthy;`

This has been corrected and the code will be reworked so it can pass the updated tests.

**Comment by ericwburden** [comment]
@brehberg Want to PR a fix to the JS tests?

**Review by ericwburden — APPROVED** [review]
@brehberg NVM my previous comment, I see you've already done so! Thanks!

**Comment by ericwburden** [comment]
Went ahead and merged to get the test fixes in.
